aws/default_retryer.go

```diff
@@ -34,25 +35,39 @@ var seededRand = rand.New(&lockedSource{src: rand.NewSource(time.Now().UnixNano(
 // RetryRules returns the delay duration before retrying this request again
 func (d DefaultRetryer) RetryRules(r *Request) time.Duration {
 	// Set the upper limit of delay in retrying at ~five minutes
-	minTime := 30
+	var minTime int64
+	if d.MinTime != 0 {
+		minTime = d.MinTime
+	} else {
+		minTime = 30
+	}
+
+	var initialDelay time.Duration
 	throttle := d.shouldThrottle(r)
 	if throttle {
-		if delay, ok := getRetryDelay(r); ok {
-			return delay
+		if delay, ok := getRetryAfterDelay(r); ok {
+			initialDelay = delay
 		}
 
 		minTime = 500
 	}
 
 	retryCount := r.RetryCount
-	if retryCount > 13 {
-		retryCount = 13
-	} else if throttle && retryCount > 8 {
+	if throttle && retryCount > 8 {
 		retryCount = 8
+	} else if retryCount > 12 {
+		retryCount = 12
 	}
 
-	delay := (1 << uint(retryCount)) * (seededRand.Intn(minTime) + minTime)
-	return time.Duration(delay) * time.Millisecond
+	maxRetriesAllowed := d.MaxRetries()
+	if maxRetriesAllowed != 0 {
+		if retryCount > maxRetriesAllowed-1 {
+			retryCount = maxRetriesAllowed - 1
+		}
+	}
+
+	delay := (1 << uint(retryCount)) * (seededRand.Int63n(minTime) + minTime)
```

Comment:
If we update the above logic to use a `MaxiumRetryDelay` instead of capping based on `retryCount` this logic will need to be updated as `retryCount` greater than 30 will have an integer overflow. (63, if the exponent is wrapped in as int64)